### ISSUE TITLE
max_nhcube=1 does not turn off stratified sampling

### ISSUE THREAD

**Issue by ndeutschmann** [gplepage/vegas#18]
Hi,

I tried using vegas as a RNG without stratified sampling by setting `max_nhcube=1` as indicated in the docs, however multiple hypercubes seem to be generated still.

```
integ = vegas.Integrator([[0,1]]*2, max_nhcube=1)
x,wx,hc = next(integ.random_batch(yield_hcube=True))
np.unique(hc)
# Returns: array([0, 1, 2, 3])
```

Am I misunderstanding how this option is supposed to work?

NB: setting `nstrat=[1]*d` does work however

**Comment by gplepage** [comment]
Thanks for reporting the problem. It is indeed a bug. I will post a new version with the bug fixed sometime this weekend. Using ``nstrat`` should work, as you report.

**Comment by gplepage** [comment]
Version 4.0.3 fixes this bug. Your code (above) now returns ``[0]`` and ``integ.settings()`` returns

```
Integrator Settings:
    1000 (approx) integrand evaluations in each of 10 iterations
    number of:  strata/axis = [1 1]
                increments/axis = [100 100]
                h-cubes = 1  evaluations/h-cube = 1000 (min)
                h-cubes/batch = 1000
    minimize_mem = False
    adapt_to_errors = False
    damping parameters: alpha = 0.5  beta= 0.75
    limits: h-cubes < 1  evaluations/h-cube < 1e+06
    accuracy: relative = 0  absolute accuracy = 0

    axis 0 covers (0.0, 1.0)
    axis 1 covers (0.0, 1.0)
```

which is as it should be. Thanks again for reporting the bug.

**Comment by gplepage** [comment]
I am closing this issue because it is addressed by v4.0.3.
